src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -375,13 +381,11 @@ private function autoPrependText(): void
         }
     }
 
-    private function reduceBuffer($messages): array
+    private function writeBuffer(string $message, bool $newLine, int $type): void
     {
         // We need to know if the two last chars are PHP_EOL
         // Preserve the last 4 chars inserted (PHP_EOL on windows is two chars) in the history buffer
-        return array_map(function ($value) {
-            return substr($value, -4);
-        }, array_merge(array($this->bufferedOutput->fetch()), (array) $messages));
```

Comment:
so the `fetch` part was useless all along?

Reply:
It wasn't useless, but as we now only handle strings one by one and are writing to the buffer directly (instead of concatenating arrays of buffered output), it is not required anymore.